src/components/modal/index.stories.mdx

```diff
@@ -0,0 +1,150 @@
+import { Canvas, Meta, Story } from '@storybook/addon-docs';
+import { action } from '@storybook/addon-actions';
+import { useDisclosure } from '@chakra-ui/react';
```

Comment:
I think you would want to expose this hook as well, no? Looks like a nice way to control the modal